VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -203,25 +207,25 @@ public void onComplete(FacebookDialog.PendingCall pendingCall, Bundle data) {
      * @param savedInstanceState
      */
     private void restore(Bundle savedInstanceState) {
-        try{
-            if (player == null) {
-                if (savedInstanceState != null
-                        && savedInstanceState.containsKey("player")) {
-                    player = (IPlayer) savedInstanceState.get("player");
-                } else {
-                    player = new Player();
-                }
-            }
+        if (player == null) {
             if (savedInstanceState != null
-                    && savedInstanceState.containsKey("isMessageDisplayed")){
-                if(savedInstanceState.getBoolean("isMessageDisplayed")){
-                    showVideoNotAvailable(VideoNotPlayMessageType.IS_VIDEO_MESSAGE_DISPLAYED);
-                }
+                    && savedInstanceState.containsKey("player")) {
+                player = (IPlayer) savedInstanceState.get("player");
+            } else {
+                player = new Player();
             }
-            reAttachPlayEventListener();
-        }catch(Exception e){
-            logger.error(e);
         }
+        if (savedInstanceState != null
+                && savedInstanceState.containsKey("isMessageDisplayed")) {
+            if (savedInstanceState.getBoolean("isMessageDisplayed")) {
+                showVideoNotAvailable(VideoNotPlayMessageType.IS_VIDEO_MESSAGE_DISPLAYED);
+            }
+        }
+        if (savedInstanceState != null) {
+            isPrepared = savedInstanceState.getBoolean("isPrepared");
+            isAutoPlayDone = savedInstanceState.getBoolean("isAutoPlayDone");
+        }
+        reAttachPlayEventListener();
```

Comment:
This is already being called from `setupController()` (which is in turn called from `handleOnResume()`), so I think it can be removed from here.